### ISSUE TITLE
ActorSystem leaks memory

### ISSUE THREAD

**Issue by isurakka** [akkadotnet/akka.net#5802]
**Version Information**
1.4.6 - 1.4.35
Did not test every single version but at least 1.4.6 and 1.4.35 leaks.
No leak on 1.4.5

.NET 6

Using Akka NuGet package

**Describe the bug**
Creating ActorSystem, terminating it and disposing it in a loop, leaves memory behind.

**To Reproduce**

Complete console app code example:
```
using Akka.Actor;

while (true)
{
    var actorSystem = ActorSystem.Create("test");

    await actorSystem.Terminate();
    actorSystem.Dispose();
    actorSystem = null;

    GC.Collect();
    GC.WaitForPendingFinalizers();
    GC.Collect();
}
```

Look at Visual Studio diagnostics memory usage or publish the app and look at memory usage in task manager. It will rise continously.

**Expected behavior**
Memory usage should flatline and not keep rising.

**Actual behavior**
Memory usage kept rising and did not stop like in 1.4.5.

**Screenshots**
1.4.35 memory usage graph: https://imgur.com/qJgGHFw
1.4.5 memory usage graph: https://imgur.com/ggyhBrc
1.4.35 memory usage snapshot diff: https://imgur.com/Mhdlwpc

**Environment**
Windows 10
.NET 6

**Additional context**

**Comment by Aaronontheweb** [comment]
Looks like the issue may have been introduced in this PR: https://github.com/akkadotnet/akka.net/pull/4402

**Comment by Arkatufus** [comment]
@isurakka the bug should be resolved in #5816
